datadog/resource_datadog_logs_pipeline.go

```diff
@@ -0,0 +1,687 @@
+				sources[i] = source
+			}
+			tfPDetails["sources"] = sources
+			tfPDetails["target"] = *urlP.Target
+			tfPDetails["normalize_ending_slashes"] = *urlP.NormalizeEndingSlashes
+		case datadog.UserAgentParserType:
+			userAgentP := ddProcessor.Definition.(datadog.UserAgentParser)
+			sources := make([]string, len(userAgentP.Sources))
+			for i, source := range userAgentP.Sources {
+				sources[i] = source
+			}
+			tfPDetails["sources"] = sources
+			tfPDetails["target"] = *userAgentP.Target
+			tfPDetails["is_encoded"] = *userAgentP.IsEncoded
+		default:
+			return fmt.Errorf("failed to support datadog processor type, %s", *ddProcessor.Type)
+		}
+		tfPDetailsList := make([]interface{}, 1, 1)
+		tfPDetailsList[0] = tfPDetails
+		tfProcessor[ddProcessorTypes[*ddProcessor.Type]] = tfPDetailsList
+		tfProcessors[i] = tfProcessor
+	}
+	return nil
+}
+
+func setProcessors(d *schema.ResourceData, processors []datadog.LogsProcessor) error {
+	tfProcessors := make([]map[string]interface{}, len(processors))
+	if err := convertDDProcessors(tfProcessors, processors); err != nil {
+		return err
+	}
+	if err := d.Set("processor", tfProcessors); err != nil {
+		return err
+	}
+	return nil
+}
+
+func buildDatadogLogsPipeline(d *schema.ResourceData) (*datadog.LogsPipeline, error) {
+	var pipeline datadog.LogsPipeline
+	pipeline.SetName(d.Get("name").(string))
+	pipeline.SetIsEnabled(d.Get("is_enabled").(bool))
+	var ddFilter = datadog.FilterConfiguration{}
+	buildFilter(d, &ddFilter)
+	pipeline.SetFilter(ddFilter)
+	processors, err := buildProcessors(d)
+	if err != nil {
+		return nil, err
+	}
+	pipeline.Processors = processors
+	return &pipeline, nil
+}
+
+func buildProcessors(d *schema.ResourceData) ([]datadog.LogsProcessor, error) {
+	tfProcessors := d.Get("processor").([]interface{})
+	ddProcessors := make([]datadog.LogsProcessor, len(tfProcessors))
+	if err := convertTFProcessors(ddProcessors, tfProcessors); err != nil {
+		return ddProcessors, err
+	}
+	return ddProcessors, nil
+}
+
+func convertTFProcessors(ddProcessors []datadog.LogsProcessor, tfProcessors []interface{}) error {
+	for i, tfEntry := range tfProcessors {
+		tfMap := tfEntry.(map[string]interface{})
+		for tfPType, ddPType := range tfProcessorTypes {
+			if v, ok := tfMap[tfPType].([]interface{}); ok && len(v) > 0 {
+				if tfP, ok := v[0].(map[string]interface{}); ok {
+					var ddProcessor = datadog.LogsProcessor{}
+					if err := convert(tfP, ddPType, &ddProcessor); err != nil {
+						return err
+					}
+					ddProcessors[i] = ddProcessor
+					break
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func convert(tfProcessor map[string]interface{}, processorType string, ddProcessor *datadog.LogsProcessor) error {
+	ddProcessor.Type = datadog.String(processorType)
+	if name, ok := tfProcessor["name"].(string); ok {
+		ddProcessor.Name = &name
+	}
+	if isEnabled, ok := tfProcessor["is_enabled"].(bool); ok {
+		ddProcessor.IsEnabled = &isEnabled
+	}
+	switch processorType {
```

Comment:
Do we want to create one method per case ?

Reply:
will split.